vmm/src/vm.rs

```diff
@@ -1557,6 +1598,50 @@ impl Vm {
         Ok(())
     }
 
+    pub fn handle_serial_pty(&self) -> Result<()> {
```

Comment:
It's a shame these two functions are basically copy paste of each other. Any thoughts on how to simplify?

Reply:
It is a little annoying. I pushed it into a single function that isn't super pretty but reduces a bit of the duplication and allows the dispatch to call just a single function.